democracy_club/templates/base.html

```diff
@@ -10,33 +10,29 @@
       <link rel="manifest" href="{% static "manifest.json" %}">
-      <meta property="fb:pages" content="316731041785" />
     {% endblock site_meta %}
-
-    <meta property="og:description" content="{% block og_description_content %}We build digital tools to support everyone’s participation in UK democracy.{% endblock og_description_content %}" />
-    <meta property="og:image" content="{{ CANONICAL_URL }}{% block og_image %}{% static 'images/logo_icon.png' %}{% endblock og_image %}" />
-    <meta property="og:title" content="{% block og_title_content %}{{ SITE_TITLE }}{% endblock og_title_content %}" />
-    <meta property="og:type" content="{% block og_type_content %}website{% endblock og_type_content %}">
-    <meta property="og:url" content="{{ CANONICAL_URL }}">
-    <meta property="og:image:width" content="{{ site_logo_width }}">
-
-    {% block base_og_tags %}
-      <meta property="fb:app_id" content="262795130596272" />
-      <meta property="og:url" content="{{ CANONICAL_URL }}" />
-      <!-- Twitter -->
-      <meta name="twitter:text:title" content="{{ SITE_TITLE }}">
-      <meta name="twitter:title" content="{{ SITE_TITLE }}">
-      <meta name="twitter:card" content="summary_large_image">
-      <meta name="twitter:site" content="@democlub">
-      <meta name="twitter:url" content="{{ CANONICAL_URL }}">
-      <meta name="twitter:image" content="{{ CANONICAL_URL }}{% static 'images/logo_icon.png' %}">
-      <meta name="twitter:description" content="{{ SITE_TITLE }}">
-      <meta name="twitter:image:alt" content="{{ SITE_LOGO_WIDTH }}">
+    {% block base_og_tags %} 
+      <meta property="og:description" content="{% block og_description_content %}We build digital tools to support everyone’s participation in UK democracy.{% endblock og_description_content %}" />
+      <meta property="og:image" content="{% block og_image_content %}{{ CANONICAL_URL }}{% static 'images/logo_icon.png' %}{% endblock og_image_content %}" />
+      <meta property="og:title" content="{% block og_title_content %}{{ SITE_TITLE }}{% endblock og_title_content %}" />
+      <meta property="og:type" content="{% block og_type_content %}website{% endblock og_type_content %}">
+      <meta property="og:url" content="{{ CANONICAL_URL }}">
 
     {% endblock base_og_tags %}
+    <!-- Facebook -->
+    <meta property="fb:pages" content="316731041785" />
+    <meta property="fb:app_id" content="262795130596272" />
+    <!-- Twitter -->
+    <meta name="twitter:title" content="{{ SITE_TITLE }}">
+    <meta name="twitter:card" content="summary_large_image">
+    <meta name="twitter:site" content="@democlub">
+    <meta name="twitter:url" content="{{ CANONICAL_URL }}">
+    <meta name="twitter:image" content="{% block og_image_twitter_content %}{{ CANONICAL_URL }}{% static 'images/maze_illustration_no_text.png' %}{% endblock og_image_twitter_content %}" />
```

Comment:
So far, to dynamically resize og:image, I've tried the following without success: 
setting a width with context processors
adding a width tag such as `<meta property="og:image:width" context="600"' or 'context="{{ site_logo-width }}"'
resizing image files or creating designated og image files
and now finally, as shown above, setting block image tag and changing images to a layout that better suits a twitter preview to be rendered dynamically.

Reply:
Update: #4 is doing what I intended; when I inspect the blog page, I see the desired image source in the og:image tag, yet the validator doesn't recognise it. I think this is due to the same domain issue we saw last week.

Reply:
I could use some advice on how to handle image source in `post-detail.html` between stage vs prod to get this working as hoped.

Reply:
So for example, in your `base.html` the image meta tag for twitter might be:
```
<meta name="twitter:image" content="{{ request.scheme }}://{{ request.get_host }}{% block og_image_content %}{% static "images/logo_icon.svg" %}"{% end block %}>
```
So that means that unless another HTML page defines the `og_image_content` block, the default of the site logo is used. But it then allows us to add this to the post_detail.html:
```
{% block og_image_content %}{% thumbnail post.hero "690" as im %}{{ im.url }}{% endthumbnail %}
{% endblock og_image_content %}
```
So that on a blog page, the image associated with the blog is used instead.

With this locally and visiting a blog page and checking the page source the full meta tag looks like this:
```
<meta name="twitter:image" content="http://localhost:8000/media/cache/24/6d/246d0a333710fe425cccd829c8851e68.jpg
">
```

Reply:
See comment below about S3 URL - which means the tag I suggested above has the `block` in the wrong place, you are right to have it right at the start of the content attribute so:
```     
<meta name="twitter:image" content="{% block og_image_twitter_content %}{ request.scheme }}://{{ request.get_host }}{% static 'images/maze_illustration_no_text.png' %}{% endblock og_image_twitter_content %}" />
```